### PR TITLE
Add sharing and registration to tiledb.cloud.asset

### PR DESCRIPTION
**Pull request by sgillies**
Starting with share() and tests.

### CODE REVIEW

**Review by Shelnutt2 — APPROVED** [review]

**Review by Shelnutt2 — APPROVED** [review]

File: src/tiledb/cloud/asset.py [hunk 1]
```diff
@@ -11,11 +11,90 @@
...
+    'your-bucket' with a prefix 'prefix' to your own namespace:
+
+    >>> from tiledb.cloud import asset
+    >>> asset.register(
+    ...     "s3://your-bucket/prefix",
+    ...     "array",
+    ...     name="new-array",
+    ... )
+    """
+    if type == "array":
```
Comment (sgillies) [inline src/tiledb/cloud/asset.py:53]: Resorted to `if/else` in this case.

File: src/tiledb/cloud/asset.py [hunk 2]
```diff
@@ -38,3 +118,109 @@ def info(uri: str) -> Union[ArrayInfo, GroupInfo]:
...
+        "array": array.list_shared_with,
+        "group": groups.list_shared_with,
+    }
+    asset_type: str = tiledb.object_type(uri, ctx=tiledb.cloud.Ctx())
+    func = sharing_map[asset_type]
+    return func(uri)
+
+
+def share(
+    uri: str, namespace: str, permissions: Optional[Union[str, list[str]]] = "read"
```
Comment (sgillies) [inline src/tiledb/cloud/asset.py:185]: I could be easily talked out of providing a default for permissions.

Comment (Shelnutt2) [reply src/tiledb/cloud/asset.py:185]: I think default here is good. Additionally at some point we need to coordinate programmatic vs ui. UI uses "roles" which they translate internally to these permissions.

File: tests/test_asset.py [hunk 3]
```diff
@@ -61,3 +73,194 @@ def test_public_group_asset_info():
...
+    "tiledb.cloud.groups.list_shared_with", return_value=[GroupSharing(namespace="bar")]
+)
+def test_asset_list_shared_with_group_dispatch(group_sharing, object_type):
+    """Dispatch to groups.info when URI is a group."""
+    sharing = asset.list_shared_with("g")
+    assert isinstance(sharing[0], GroupSharing)
+    assert sharing[0].namespace == "bar"
+
+
+@pytest.mark.xfail(reason="Server error 610")
```
Comment (sgillies) [inline tests/test_asset.py:124]: Oddly there is a server error in this case, but not with the same test of a public group.

**Comment by sgillies** [comment]
@JohnMoutafis how does this look to you? It completes the new asset.py module.

**Review by JohnMoutafis — APPROVED** [review]

**Comment by JohnMoutafis** [comment]
@sgillies Is this ready to merge or does it need more work?

**Comment by sgillies** [comment]
@JohnMoutafis this is good to go. I'll merge now.

**Comment by JohnMoutafis** [comment]
Thank you @sgillies

**Pull request merged**